tests/unit/optimization/FirstOrderSAGCostFunctionInterface_unittest.cc

```diff
@@ -0,0 +1,184 @@
+/*
+* Copyright (c) The Shogun Machine Learning Toolbox
+* Written (W) 2018 Elfarouk
+* All rights reserved.
+*
+* Redistribution and use in source and binary forms, with or without
+* modification, are permitted provided that the following conditions are met:
+*
+* 1. Redistributions of source code must retain the above copyright notice, this
+*    list of conditions and the following disclaimer.
+* 2. Redistributions in binary form must reproduce the above copyright notice,
+*    this list of conditions and the following disclaimer in the documentation
+*    and/or other materials provided with the distribution.
+*
+* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+* ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+* WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+* DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+* ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+* (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+* LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+* ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+* (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+* SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*
+* The views and conclusions contained in the software and documentation are those
+* of the authors and should not be interpreted as representing official policies,
+* either expressed or implied, of the Shogun Development Team.
+*
+*/
+
+#include <gtest/gtest.h>
+
+#include <shogun/optimization/FirstOrderSAGCostFunctionInterface.h>
+#include "FirstOrderSAGCostFunctionInterface_unittest.h"
+#include <Eigen/Dense>
+
+using namespace shogun;
+using namespace Eigen;
+
+/** This is a temporary fix. The variables are now stan::math::variables
+*   So we cannot return a reference to the minimizer in the format
+*   SGVector< stan::math::Var > until we update the minimizer's interface
+*/
+SGVector<float64_t>  LeastSquareTestCostFunction::obtain_variable_reference()
+{
+  int32_t params_num = m_X->cols();
+  SGVector<float64_t> ret(params_num);
+  for(auto i=0; i<params_num; ++i)
+  {
+    ret[i] = (*m_trainable_parameters)[i].val();
+  }
+  return ret;
+}
+
+
+TEST(LeastSquareTestCostFunction, ONALINE)
```

Comment:
whats ONALINE?
Pls try to do slightly more elaborate test names :)

Reply:
Sorry, meant that the points are on a line (ONALINE)

Reply:
points_on_a_line